Integrations/AWS-CloudTrail/AWS-CloudTrail.py

```diff
@@ -0,0 +1,400 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+import boto3
+import datetime
+
+AWS_DEFAULT_REGION = demisto.params()['defaultRegion']
+AWS_roleArn = demisto.params()['roleArn']
+AWS_roleSessionName = demisto.params()['roleSessionName']
+AWS_roleSessionDuration = demisto.params()['sessionDuration']
+AWS_rolePolicy = None
+
+def aws_session(service='cloudtrail',region=None,roleArn=None,roleSessionName=None,roleSessionDuration=None,rolePolicy=None):
+    kwargs = {}
+    if roleArn and roleSessionName is not None:
+        kwargs.update({
+            'RoleArn':roleArn,
+            'RoleSessionName':roleSessionName,
+            })
+    elif AWS_roleArn and AWS_roleSessionName is not None:
+        kwargs.update({
+            'RoleArn':AWS_roleArn,
+            'RoleSessionName':AWS_roleSessionName,
+            })
+
+    if roleSessionDuration is not None:
+        kwargs.update({'DurationSeconds':int(roleSessionDuration)})
+    elif AWS_roleSessionDuration is not None:
+        kwargs.update({'DurationSeconds':int(AWS_roleSessionDuration)})
+
+    if rolePolicy is not None:
+        kwargs.update({'Policy':rolePolicy})
+    elif AWS_rolePolicy is not None:
+        kwargs.update({'Policy':AWS_rolePolicy})
+
+    if kwargs:
+        sts_client = boto3.client('sts')
+        sts_response = sts_client.assume_role(**kwargs)
+        if region is not None:
+            client = boto3.client(
+                service_name=service,
+                region_name=region,
+                aws_access_key_id=sts_response['Credentials']['AccessKeyId'],
+                aws_secret_access_key=sts_response['Credentials']['SecretAccessKey'],
+                aws_session_token=sts_response['Credentials']['SessionToken']
+                )
+        else:
+            client = boto3.client(
+                service_name=service,
+                region_name=AWS_DEFAULT_REGION,
+                aws_access_key_id=sts_response['Credentials']['AccessKeyId'],
+                aws_secret_access_key=sts_response['Credentials']['SecretAccessKey'],
+                aws_session_token=sts_response['Credentials']['SessionToken']
+                )
+    else:
+        if region is not None:
+            client = boto3.client(service_name=service,region_name=region)
+        else:
+            client = boto3.client(service_name=service,region_name=AWS_DEFAULT_REGION)
+
+    return client
```

Comment:
Let's replace with the new client so that we can support access keys/ and proxies.

```python
def aws_session(service='cloudtrail', region=None, roleArn=None, roleSessionName=None, roleSessionDuration=None,
                rolePolicy=None):
    kwargs = {}
    if roleArn and roleSessionName is not None:
        kwargs.update({
            'RoleArn': roleArn,
            'RoleSessionName': roleSessionName,
        })
    elif AWS_ROLE_ARN and AWS_ROLE_SESSION_NAME is not None:
        kwargs.update({
            'RoleArn': AWS_ROLE_ARN,
            'RoleSessionName': AWS_ROLE_SESSION_NAME,
        })

    if roleSessionDuration is not None:
        kwargs.update({'DurationSeconds': int(roleSessionDuration)})
    elif AWS_ROLE_SESSION_DURATION is not None:
        kwargs.update({'DurationSeconds': int(AWS_ROLE_SESSION_DURATION)})

    if rolePolicy is not None:
        kwargs.update({'Policy': rolePolicy})
    elif AWS_ROLE_POLICY is not None:
        kwargs.update({'Policy': AWS_ROLE_POLICY})
    if kwargs and AWS_ACCESS_KEY_ID is None:

        if AWS_ACCESS_KEY_ID is None:
            sts_client = boto3.client('sts', config=config, verify=VERIFY_CERTIFICATE)
            sts_response = sts_client.assume_role(**kwargs)
            if region is not None:
                client = boto3.client(
                    service_name=service,
                    region_name=region,
                    aws_access_key_id=sts_response['Credentials']['AccessKeyId'],
                    aws_secret_access_key=sts_response['Credentials']['SecretAccessKey'],
                    aws_session_token=sts_response['Credentials']['SessionToken'],
                    verify=VERIFY_CERTIFICATE,
                    config=config
                )
            else:
                client = boto3.client(
                    service_name=service,
                    region_name=AWS_DEFAULT_REGION,
                    aws_access_key_id=sts_response['Credentials']['AccessKeyId'],
                    aws_secret_access_key=sts_response['Credentials']['SecretAccessKey'],
                    aws_session_token=sts_response['Credentials']['SessionToken'],
                    verify=VERIFY_CERTIFICATE,
                    config=config
                )
    elif AWS_ACCESS_KEY_ID and AWS_ROLE_ARN:
        sts_client = boto3.client(
            service_name='sts',
            aws_access_key_id=AWS_ACCESS_KEY_ID,
            aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
            verify=VERIFY_CERTIFICATE,
            config=config
        )
        kwargs.update({
            'RoleArn': AWS_ROLE_ARN,
            'RoleSessionName': AWS_ROLE_SESSION_NAME,
        })
        sts_response = sts_client.assume_role(**kwargs)
        client = boto3.client(
            service_name=service,
            region_name=AWS_DEFAULT_REGION,
            aws_access_key_id=sts_response['Credentials']['AccessKeyId'],
            aws_secret_access_key=sts_response['Credentials']['SecretAccessKey'],
            aws_session_token=sts_response['Credentials']['SessionToken'],
            verify=VERIFY_CERTIFICATE,
            config=config
        )
    else:
        if region is not None:
            client = boto3.client(
                service_name=service,
                region_name=region,
                aws_access_key_id=AWS_ACCESS_KEY_ID,
                aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
                verify=VERIFY_CERTIFICATE,
                config=config
            )
        else:
            client = boto3.client(
                service_name=service,
                region_name=AWS_DEFAULT_REGION,
                aws_access_key_id=AWS_ACCESS_KEY_ID,
                aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
                verify=VERIFY_CERTIFICATE,
                config=config
            )

    return client
```